ui/src/app/domains/views/DomainDetails/ResourceRecords/DeleteRecordForm/DeleteRecordForm.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback } from "react";
+
+import { Icon } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikForm from "app/base/components/FormikForm";
+import type { EmptyObject } from "app/base/types";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+import type { Domain, DomainResource } from "app/store/domain/types";
+import { isDomainDetails } from "app/store/domain/utils";
+import type { RootState } from "app/store/root/types";
+
+type Props = {
+  closeForm: () => void;
+  id: Domain["id"];
+  resource: DomainResource;
+};
+
+const DeleteRecordForm = ({
+  closeForm,
+  id,
+  resource,
+}: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const domain = useSelector((state: RootState) =>
+    domainSelectors.getById(state, id)
+  );
+  const errors = useSelector(domainSelectors.errors);
+  const saved = useSelector(domainSelectors.saved);
+  const saving = useSelector(domainSelectors.saving);
+  const cleanup = useCallback(() => domainActions.cleanup(), []);
+
+  if (!isDomainDetails(domain)) {
+    return null;
+  }
+
+  const hasMultipleRecords =
+    domain.rrsets.filter(
+      (rrset) => rrset.dnsresource_id === resource.dnsresource_id
+    ).length > 1;
+
+  return (
+    <FormikForm<EmptyObject>
```

Comment:
Should this use the `TableDeleteConfirm` component?

Reply:
I tried using `TableDeleteConfirm` but I was having issues getting the errors to show up properly, since it seems to handle it differently. I feel like it should use `FormikForm` as a base so that it's the same.

Reply:
Sure, thanks.